File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/turn/activity/EveryoneFoundActivity.java
package fr.erban.dxitcompanion.game.turn.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import fr.erban.dxitcompanion.R;
import fr.erban.dxitcompanion.game.GameBean;
import fr.erban.dxitcompanion.game.turn.Turn;

public class EveryoneFoundActivity extends AppCompatActivity {

    private Turn turn;

    private GameBean gameBean;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.everyone_found);

        this.turn = (Turn) getIntent().getSerializableExtra("Turn");
        this.gameBean = (GameBean) getIntent().getSerializableExtra("Game");

        final TextView turnNumber = findViewById(R.id.turnNumber);
        final String turnNumberComplete = getString(R.string.turnNumberPrefix) + gameBean.getCurrentTurn();
        turnNumber.setText(turnNumberComplete);
    }

    public void clickYes(View view) {

        this.turn.setEverybodyFound(true);
        endTurn();
    }

    public void clickNo(View view) {

        this.turn.setEverybodyFound(false);
        toWhoDidFind();
    }

    private void toWhoDidFind() {

        Intent intent = new Intent(EveryoneFoundActivity.this, WhoDidFindActivity.class);
        intent.putExtra("Game", gameBean);
        intent.putExtra("Turn", turn);
        EveryoneFoundActivity.this.startActivity(intent);
    }

    private void endTurn() {

        Intent intent = new Intent(EveryoneFoundActivity.this, EndTurnActivity.class);
        intent.putExtra("Game", gameBean);
        intent.putExtra("Turn", turn);
        EveryoneFoundActivity.this.startActivity(intent);
    }
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/turn/activity/EveryoneFoundActivity.kt
package fr.erban.dxitcompanion.game.turn.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import fr.erban.dxitcompanion.R
import fr.erban.dxitcompanion.common.parcelableExtra
import fr.erban.dxitcompanion.databinding.EveryoneFoundBinding
import fr.erban.dxitcompanion.game.GameBean
import fr.erban.dxitcompanion.game.turn.Turn

class EveryoneFoundActivity : AppCompatActivity() {

    private lateinit var binding: EveryoneFoundBinding
    private lateinit var gameBean: GameBean
    private lateinit var turn: Turn

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = EveryoneFoundBinding.inflate(layoutInflater)
        setContentView(binding.root)

        turn = intent.parcelableExtra("Turn")!!
        gameBean = intent.parcelableExtra("Game")!!
        binding.turnNumber.text = getString(R.string.turnNumberPrefix) + gameBean.currentTurn
    }

    fun clickYes(view: View) {
        turn = turn.copy(everybodyFound = true)
        startActivity(
            Intent(this, EndTurnActivity::class.java)
                .putExtra("Game", gameBean)
                .putExtra("Turn", turn)
        )
    }

    fun clickNo(view: View) {
        turn = turn.copy(everybodyFound = false)
        startActivity(
            Intent(this, WhoDidFindActivity::class.java)
                .putExtra("Game", gameBean)
                .putExtra("Turn", turn)
        )
    }
}
